clang/include/clang/AST/Type.h

```diff
@@ -6155,6 +6155,53 @@ class BTFTagAttributedType : public Type, public llvm::FoldingSetNode {
   }
 };
 
+class HLSLAttributedResourceType : public Type, public llvm::FoldingSetNode {
+public:
+  struct Attributes {
+    // Data gathered from HLSL resource attributes
+    uint8_t ResourceClass; // maps to llvm::dxil::ResourceClass
+    uint8_t IsROV : 1;
+    Attributes(uint8_t RC, bool rov) : ResourceClass(RC), IsROV(rov) {}
```

Comment:
nit, "rov" should be capitalized. It's fairly common practice to just shadow the member names in simple constructors like this, so I might actually write it like so:
```suggestion
    Attributes(uint8_t ResourceClass, bool IsROV)
        : ResourceClass(ResourceClass), IsROV(IsROV) {}
```